qiskit/algorithms/linear_solvers/observables/absolute_average.py

```diff
@@ -0,0 +1,89 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The absolute value of the average of the components of the vector solution to the
+linear systems."""
+from typing import Union, Optional, List
+import numpy as np
+
+from qiskit.aqua.operators import PauliSumOp
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit.opflow import I, Z, TensoredOp
+from .linear_system_observable import LinearSystemObservable
+
+
+class AbsoluteAverage(LinearSystemObservable):
+    r"""A class for the absolute value of the average of the components of the vector solution
+    to the linear systems.
+    For a vector :math:`x=(x_1,...,x_N)`, the average is defined as
+    :math:`\abs{\frac{1}{N}\sum_{i-1}^{N}x_i}`."""
+    def observable(self, num_qubits: int) -> Union[PauliSumOp, List[PauliSumOp]]:
```

Comment:
This states is returns a Union of PauliSumOp or List thereof.m but is in fact returning just a TensoredOp. There is some mistmatch here that needs reconciling

Reply:
Changed to TensoredOp also in the signature of the `linear_system_observable` class, this was a late change to speed up the running time of the algorithm (~5min using PauliSumOp compared to a few seconds when casting the observable as a TensoredOp). The reason it is an Union with a List is to match the signature from the parent class because e.g. the `matrix_function` observable returns a List of TensoredOp's.